### PR TITLE
added arangodb backend

### PR DESCRIPTION
**Pull request by dilipvamsi**
### Backend url
_**arangodb://username:password@host:port/database/collection**_
### Custom Backend settings 
**arangodb_backend_settings** is where the settings are present in the app.conf.
Settings should contain the **host, port, username, password, database name, collection name** else the default will be chosen. 
## Task Document in DB
Task is written into the collection with data in task attribute and task_id as _key (document key)
### Sample Document
```json
  {
    "_key": "celery-task-meta-820dd23a-d16e-4527-b15b-aaeb145857bf",
    "_id": "celery/celery-task-meta-820dd23a-d16e-4527-b15b-aaeb145857bf",
    "_rev": "_X1isPje--_",
    "task": {
      "status": "SUCCESS",
      "result": 7,
      "traceback": null,
      "children": [],
      "task_id": "820dd23a-d16e-4527-b15b-aaeb145857bf",
      "date_done": "2018-12-04T12:25:34.709131"
    }
  }
```

Regards,
Dilip Vamsi.

### CODE REVIEW

File: celery/backends/arangodb.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+"""Couchbase result store backend."""
```
Comment (clokep) [inline celery/backends/arangodb.py]: Is this a copy and paste error? (Couchbase vs. ArangoDB?)

Comment (dilipvamsi) [reply celery/backends/arangodb.py]: Yes.

**Comment by dilipvamsi** [comment]
Hi @clokep ,
Can you help me for the required
-  Test Cases
-  Proper Documentation for integration with existing docs.

Thanks,
Dilip Vamsi.

File: celery/app/backends.py [hunk 2]
```diff
@@ -37,6 +37,7 @@
     'consul': 'celery.backends.consul:ConsulBackend',
     'dynamodb': 'celery.backends.dynamodb:DynamoDBBackend',
     'azureblockblob': 'celery.backends.azureblockblob:AzureBlockBlobBackend',
+    'arangodb': 'celery.backends.arangodb.ArangoDbBackend'
```
Comment (clokep) [inline celery/app/backends.py]: I think you need a `:` before `ArangoDbBackend` instead of a `.`.

Also, please add a trailing comma here.

**Comment by clokep** [comment]
@dilipvamsi I've not added a backend, so not 100% sure what test-cases are necessary! Sorry. I know there was a couple of other PRs out recently about this though. You might derive inspiration from them! For docs, I'd look at the same thing (or try to make similar documentation as what exists for other backends).

File: celery/app/backends.py (hunk 2 above)
Comment (dilipvamsi) [reply celery/app/backends.py]: I didn't get the first comment

Comment (dilipvamsi) [reply celery/app/backends.py]: I have added the trailing comma

Comment (clokep) [reply celery/app/backends.py]: @dilipvamsi You have `celery.backends.arangodb.ArangoDbBackend` (everything separated by dots), but if you look at the other backends, the class-name (e.g. `ArangoDbBackend`) is separated by the module via a colon, so it should be `celery.backends.arangodb:ArangoDbBackend`. Subtle, but I believe it is what lets the dynamic importing work properly.

File: celery/backends/arangodb.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+        if url is None:
+            host = port = database = collection = username = password = None
+        else:
+            (
+                _schema, host, port, username, password,
+                database_collection, _query
+            ) = _parse_url(url)
+            logging.debug(
+                "_shema: %s, host: %s, port: %s, username: %s, password: %s, "
```
Comment (thedrow) [inline celery/backends/arangodb.py]: Never log passwords.
I'd rather remove that line entirely anyway.

File: celery/backends/arangodb.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+import logging
+import json
+from kombu.utils.url import _parse_url
+
+from celery.exceptions import ImproperlyConfigured
+
+from .base import KeyValueStoreBackend
+
+try:
+    from pyArango import connection as pyArangoConnection
```
Comment (thedrow) [inline celery/backends/arangodb.py]: Module name should be name_with_underscores.

**Comment by dilipvamsi** [comment]
Hi @thedrow ,
Can you explain about the `bandit.json` file and whats needed to be done in that file.
Thanks,
Dilip Vamsi

**Comment by dilipvamsi** [comment]
Hi, @thedrow ,
I have taken example of couchbase and written the documentation. But it seems that I'm missing something as sphinx build is failing. Can you look into it and help me for the required.

Thanks,
Dilip Vamsi.

File: celery/backends/arangodb.py [hunk 5]
```diff
@@ -60,7 +60,7 @@ def __init__(self, url=None, *args, **kwargs):
                 database_collection, _query
             ) = _parse_url(url)
             logging.debug(
```
Comment (thedrow) [inline celery/backends/arangodb.py]: Why do we need this log line?

Comment (dilipvamsi) [reply celery/backends/arangodb.py]: Its just the configuration of the backend.
Are there any strict guidelines regarding the logs as I'm even logging on a get, set, mget and delete

**Comment by dilipvamsi** [comment]
Hi @thedrow @clokep @georgepsarakis @auvipy,
I have fixed the sphinx documentation and added the arangodb test cases (all passing when I locally ran them).
But Travis CI seems to fail because of the setup error as shown below.
Please update and let me know.
setup error

Thanks,
Dilip Vamsi.

**Comment by thedrow** [comment]
If you could, please add integration tests.

**Comment by dilipvamsi** [comment]
Hi @thedrow @auvipy ,
Can you suggest me what needs to be implemented in integration testing.
Thanks,
Dilip Vamsi.

**Comment by thedrow** [comment]
We just need to run ArangoDB in CI.
So include it in tox and in the build matrix.

**Pull request merged**